spec/models/reservation_spec.rb

```diff
@@ -87,705 +73,830 @@
+    let!(:archiver) { FactoryGirl.build_stubbed(:user) }
+    it 'returns the reservation' do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      expect(res.archive(archiver, '')).to eq(res)
+    end
+    context 'not checked in' do
+      context 'not checked out' do
+        it 'sets checked_out to now' do
+          travel(-1.days) do
+            res = FactoryGirl.build_stubbed(:valid_reservation)
+            expect { res.archive(archiver, '') }.to change { res.checked_out }
+              .from(nil).to(Time.zone.now)
+          end
+        end
+      end
+      it 'sets checked_in to now' do
+        travel(-1.days) do
+          res = FactoryGirl.build_stubbed(:checked_out_reservation)
+          expect { res.archive(archiver, '') }.to change { res.checked_in }
+            .from(nil).to(Time.zone.now)
+        end
+      end
+      it 'updates the notes' do
+        travel(-1.days) do
+          res = FactoryGirl.build_stubbed(:checked_out_reservation)
+          allow(archiver).to receive(:md_link).and_return('md link')
+          note = 'note'
+          archive_time = Time.zone.now.to_s(:long)
+          res.archive(archiver, note)
+          expect(res.notes).to include(note)
+          expect(res.notes).to include('md link')
+          expect(res.notes).to include(archive_time)
+        end
+      end
+      it 'sets the status to archived' do
+        res = FactoryGirl.build_stubbed(:checked_out_reservation)
+        expect { res.archive(archiver, '') }.to change { res.status }
+          .from('checked_out').to('archived')
+      end
     end
-
-    it 'returns the full amount if no cap set' do
-      reservation.equipment_model.update_attributes(late_fee: 100,
-                                                    late_fee_max: 0)
-      reservation.update_attributes(due_date: Time.zone.today - 3.days)
-      expected = (Time.zone.today - reservation.due_date) *
-                 reservation.equipment_model.late_fee
-      expect(reservation.late_fee).to eq(expected)
+    context 'checked in' do
+      it 'does nothing' do
+        res = FactoryGirl.build_stubbed(:checked_in_reservation)
+        expect { res.archive(archiver, '') }.not_to change { res.status }
+      end
     end
   end
 
-  context '#end_date' do
-    context 'if checked in' do
-      before { @res = FactoryGirl.build(:checked_in_reservation) }
-
-      it 'returns the checkin date' do
-        expect(@res.end_date).to eq(@res.checked_in)
+  describe '#update' do
+    let!(:user) { FactoryGirl.build_stubbed(:user) }
+    it 'does nothing when no changes and no new notes' do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      expect { res.update(user, {}, '') }.not_to change { res.notes }
+    end
+    it 'updates with given notes' do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      res.update(user, {}, 'test_note')
+      expect(res.notes).to include('test_note')
+    end
+    it 'notes the editing user' do
```

Comment:
We can leave these for now but ultimately I think we'll be better off just testing the call to whatever `ReservationNoter` object we create.